main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"fmt"
+)
+
+type item struct {
+	name   string
+	amount int
+	price  int
+}
+
+func main() {
+	//crio os items
+	item1 := item{
+		name:   "batata",
+		amount: 0,
+		price:  100,
+	}
+	item2 := item{
+		name:   "tomate",
+		amount: 10,
+		price:  15,
+	}
+	//crio a lista de items
+	itemList := map[int]item{
+		1: item1,
+		2: item2,
+	}
+	//printo a lista
+	//fmt.Println("Lista de items: ", itemList)
+
+	//crio a lista de e-mails
+	emailList := []string{
+		"mary@mail.com",
+		"john@mail.com",
+		"rose@mail.com",
+		"sunny@mail.com",
+		"julia@mail.com",
+		"debra@mail.com",
+		"adam@mail.com",
+	}
+	//printo a lista
+	//fmt.Println("Lista de emails: ", emailList)
+	DivideTab(itemList, emailList)
+
+}
+
+//função que calcula a conta, recebe os itens e os clientes
+func DivideTab(il map[int]item, el []string) (map[string]int, error) {
+	total := 0
+	tab := make(map[string]int, len(el))
+
+	if verifyItems(il) != nil {
+		return verifyItems(il), nil
```

Comment:
Essa forma de usar a função ficou um pouco estranha, tendo que repetir ela. Pelo que eu entendi você usa ela para verificar se a lista tem algum problema. O que acha de fazer dessa forma?

A ideia é que sua função `verifyItems` retorne um erro ao invés de um map, já que ela é usada apenas para validar, aproveitando também para mudar a nomenclatura para validate para se aproximar do que de fato ela faz.

```suggestion
	if err := validateitems(il); err != nil {
		return nil, err
```